examples/reminders/lib/feature_dashboard/blocs/dashboard_bloc_extensions.dart

```diff
@@ -8,3 +8,71 @@ extension _ToError on Stream<Exception> {
   /// TODO: Implement error event-to-state logic
   Stream<String> toMessage() => map((errorState) => errorState.toString());
 }
+
+extension _DashboardList on Stream<List<ReminderModel>> {
+  Stream<ReminderModel?> _dashboardUpdatedReminder(_dashboardModelResult) =>
+      withLatestFrom(
+          _dashboardModelResult,
+          (List<ReminderModel> updatedList, Result<DashboardModel> oldList) =>
+              _UpdatedAndOldList(
+                  updatedList: updatedList,
+                  oldList: oldList)).switchMap((event) {
+        final _updatedList = event.updatedList;
+        final _oldList =
+            (event.oldList as ResultSuccess<DashboardModel>).data.reminderList;
+        ReminderModel? result;
+        if (_oldList.length > _updatedList.length) {
+          result =
+              _setInResultAReminderFromOldList(_oldList, _updatedList, result);
+        } else if (_oldList.length < _updatedList.length) {
+          result = _setInResultAReminderFromUpdatedList(
+              _updatedList, _oldList, result);
+        }
+        return Stream.value(result);
+      });
+}
+
+ReminderModel? _setInResultAReminderFromUpdatedList(
```

Comment:
As discussed this solution is overcomplicated. Please refactor it to a more simple one